main.nf

```diff
@@ -333,7 +333,7 @@ if (!params.references_parsed && !params.skip_mirdeep){
     sed -i 's, ,_,g' \$MATURE
 
     # Build bowtie index
-    bowtie-build genome.fa genome
+    bowtie-build genome.fa genome --threads \$(nproc) 
```

Comment:
Auto-detecting the number of CPUs is generally a bad idea as it is not always portable across systems and can cause issues on large HPCs where jobs share nodes etc.

Nextflow knows how many cpus should be assigned to a given task, so better to use that instead - `task.cpus`.

```suggestion
    bowtie-build genome.fa genome --threads ${task.cpus}
```